p2p/protocol/identify/id.go

```diff
@@ -349,7 +349,7 @@ func (ids *IDService) identifyConn(c network.Conn, signal chan struct{}) {
 		}
 	}()
 
-	s, err = c.NewStream()
+	s, err = c.NewStream(context.TODO())
```

Comment:
@aarshkshah1992 FYI. In your identify refactor, please consider passing in a context here.